packages/grpc-web/index.d.ts

```diff
@@ -1,6 +1,9 @@
 declare module "grpc-web" {
 
   export interface Metadata { [s: string]: string; }
+  export type StatusMetadata = Metadata & {

```

Comment:
I appreciate the type extension, and i personally prefer this kind of change.

However, since I plan to have this feature only in open source grpc-web, i prefer to make minimal changes to the rest of the API (e.g. in `RpcError`, etc.). Would it be possible if you just modify Metadata instead?

Reply:
That was my first tought, but, Metadata type is used both for input and output to grpc-services. Also i assume that previous definition od Metadata follows this: https://grpc.io/docs/guides/metadata/#overview (keys are string, values are string or binary).

To not make a breaking change for those who rely on `Record<string, string>` i created separate Metadata for request statuses (status + error) that additionaly can have new `httpStatusCode` property of number type.

I think this is more secure way to handle Metadata for status and error.

Don't see any reason why `httpStatusCode` should be valid property for standard messaging (in-out).

That's why i decided to make separate type.